### PR TITLE
refactor for the return type and necessary things

### PR DESCRIPTION
**Pull request by habibdevsy**
- refactor for the return type and necessary things

### CODE REVIEW

File: backend/src/Controller/Subscription/SubscriptionController.php [hunk 1]
```diff
@@ -23,16 +23,17 @@
...
-    /**
-     * @param SerializerInterface $serializer
-     * @param AutoMapping $autoMapping
-     * @param ValidatorInterface $validator
-     * @param SubscriptionService $subscriptionService
-     */
-    public function __construct( SerializerInterface $serializer, private AutoMapping $autoMapping,
-                                private ValidatorInterface  $validator, private SubscriptionService $subscriptionService)
+    private $autoMapping;
+    private $validator;
```
Comment (ralsalloum) [inline backend/src/Controller/Subscription/SubscriptionController.php]: Class parameters' types need to be declared.

File: backend/src/Manager/Subscription/SubscriptionManager.php [hunk 2]
```diff
@@ -19,23 +19,26 @@
...
-     * @param \App\Manager\Subscription\SubscriptionHistoryManager $subscriptionHistoryManager
-     */
-    public function __construct(private AutoMapping $autoMapping, private EntityManagerInterface $entityManager, private SubscriptionEntityRepository $subscribeRepository, private PackageManager $packageManager, private StoreOwnerProfileManager $storeOwnerProfileManager, private SubscriptionDetailsManager $subscriptionDetailsManager, private SubscriptionHistoryManager $subscriptionHistoryManager)
+    private $autoMapping;
+    private $entityManager;
+    private $packageManager;
+    private $storeOwnerProfileManager;
+    private $subscriptionDetailsManager;
+    private $subscriptionHistoryManager;
+    private $subscribeRepository;
```
Comment (ralsalloum) [inline backend/src/Manager/Subscription/SubscriptionManager.php]: Class parameters' types need to be declared.

File: backend/src/Service/Subscription/SubscriptionService.php [hunk 3]
```diff
@@ -10,25 +10,19 @@
...
 
 class SubscriptionService
 {
-    /**
-     * @param AutoMapping $autoMapping
-     * @param SubscriptionManager $subscriptionManager
-     */
-    public function __construct(private AutoMapping $autoMapping, private SubscriptionManager $subscriptionManager)
+    private $autoMapping;
+    private $subscriptionManager;
```
Comment (ralsalloum) [inline backend/src/Service/Subscription/SubscriptionService.php]: Class parameters' types need to be declared.

File: backend/src/Service/Subscription/SubscriptionService.php [hunk 4]
```diff
@@ -42,12 +36,7 @@ public function createSubscription(SubscriptionCreateRequest $request): mixed
         return $this->autoMapping->map(SubscriptionEntity::class, SubscriptionResponse::class, $subscription);
     }
 
-    /**
-     * Check if there is a future subscription
-     * @param $storeOwner
-     * @return INT
-     */
-    public function getIsFutureState($storeOwner): INT
+    public function getIsFutureState(int $storeOwner): INT
```
Comment (ralsalloum) [inline backend/src/Service/Subscription/SubscriptionService.php]: Why INT not int?

**Review by ralsalloum — CHANGES_REQUESTED** [review]
Please resolve the request changes.

**Comment by habibdevsy** [comment]
- resolve the requested changes is done.

**Review by ralsalloum — APPROVED** [review]

**Pull request merged**